app/src/main/java/com/greenfox/fuchsit/zerdareader/activity/LoginActivity.java

```diff
@@ -59,27 +62,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
     public void login(View view) {
 
-        username = editUserName.getText().toString();
-        password = editPassword.getText().toString();
-        Call<UserResponse> call = apiService.loginUser(username, password);
+        Call<UserResponse> call = apiService.loginUser(editEmail.getText().toString(), editPassword.getText().toString());
 
         call.enqueue(new Callback<UserResponse>() {
             @Override
             public void onResponse(Call<UserResponse> call, Response<UserResponse> response) {
                 UserResponse user = response.body();
 
-                loginData = PreferenceManager.getDefaultSharedPreferences(LoginActivity.this);
-                final SharedPreferences.Editor editor = loginData.edit();
-
-                editor.putString("userName", username);
-                editor.putString("password", password);
-                editor.putBoolean("isLogin", true);
-                editor.apply();
-
-                Toast.makeText(LoginActivity.this, "Saved", Toast.LENGTH_LONG).show();
-
-                Intent i = new Intent(LoginActivity.this, MainActivity.class);
-                startActivity(i);
+                checkCredentialsAndLogIn();
```

Comment:
@vikukzs checking if the editTexts are empty should be before sending an "empty" request to the server